### PR TITLE
Use Thrust and remove the dependency of CUB

### PR DESCRIPTION
**Pull request by MelodyShih**
I am able to find all the corresponding functions in Thrust and used them to replace the calls to functions in CUB. These commits removed the dependency of CUB.

### CODE REVIEW

File: src/2d/spread2d_wrapper.cu [hunk 1]
```diff
@@ -65,6 +65,15 @@ int cufinufft_spread2d(int ms, int mt, int nf1, int nf2, CPX* h_fw, int M,
 		2*M*sizeof(FLT)+M*sizeof(CUCPX), milliseconds);
 #endif
 
+	if(d_plan->opts.gpu_method == 1){
```
Comment (janden) [inline src/2d/spread2d_wrapper.cu:68]: So this is not related to CUB/Thrust, right? It's fine to include it in this PR, but would be good to separate these things in the future.

**Comment by janden** [comment]
Thank you for doing this. Should simplify dependencies and the overall code.

**Comment by garrettwrong** [comment]
> Nice, thanks! Code looks lighter. I'll try to give it a closer look and build/run it tomorrow (daughter just got home). Thanks for doing this.

The library builds, and the tests run for me.  I get similar results to master for the 2d and 3d tests.  I have a question about the 3d test, but it is unrelated to this PR.

File: src/2d/spread2d_wrapper.cu (hunk 1 above)
Comment (MelodyShih) [reply src/2d/spread2d_wrapper.cu:68]: You're right. That change is not actually related to this pull request. Ok, I will do that in the future.

**Pull request merged**